plugin-enclave-gradle/build.gradle

```diff
@@ -1,16 +1,10 @@
+//  Note: the settings specifying the mapping of this plugin
+//    to its implementation class are in the root build.gradle
+
 plugins {
-    id 'java-gradle-plugin'
     id 'org.jetbrains.kotlin.jvm'
 }
 
-gradlePlugin {
```

Comment:
Is it necessary to move this? This configuring this module to be a Gradle plugin, which is a property independent of publishing, and thus is best placed here.

Reply:
Not strictly, but it needs `java-gradle-plugin` which I moved to avoid collateral issues as `java-gradle-plugin` interacts with `maven-publish` to generate the publications for the plugin.

If we want to keep this here, I would still need `java-gradle-plugin` in the root `build.gradle`. And I would need to put a comement here

Reply:
>If we want to keep this here, I would still need java-gradle-plugin in the root build.gradle

I think that's OK.

Reply:
Let me put it back, will write the comment

Reply:
Should be fine now